modules/planning/on_lane_planning.cc

```diff
@@ -486,5 +500,176 @@ bool OnLanePlanning::CheckPlanningConfig(const PlanningConfig& config) {
   return true;
 }
 
+void AddStGraph(const STGraphDebug& st_graph, Chart* chart) {
+  chart->set_title(st_graph.name());
+  auto* options = chart->mutable_options();
+  options->mutable_x()->set_min(-2.0);
+  options->mutable_x()->set_max(10.0);
+  options->mutable_x()->set_label_string("t(second)");
+  options->mutable_y()->set_label_string("s(meter)");
+  options->set_legend_display(false);
+  for (const auto& boundary : st_graph.boundary()) {
+    auto* boundary_chart = chart->add_polygon();
+
+    // from 'ST_BOUNDARY_TYPE_' to the end
+    std::string type =
+        StGraphBoundaryDebug_StBoundaryType_Name(boundary.type()).substr(17);
+    boundary_chart->set_label(boundary.name() + "_" + type);
+    for (const auto& point : boundary.point()) {
+      auto* point_debug = boundary_chart->add_point();
+      point_debug->set_x(point.t());
+      point_debug->set_y(point.s());
+    }
+  }
+}
+
+void AddSlFrame(const SLFrameDebug& sl_frame, Chart* chart) {
+  chart->set_title(sl_frame.name());
+  auto* options = chart->mutable_options();
+  options->mutable_x()->set_min(0.0);
+  options->mutable_x()->set_max(80.0);
+  options->mutable_x()->set_label_string("s (meter)");
+  options->mutable_y()->set_min(-8.0);
+  options->mutable_y()->set_max(8.0);
+  options->mutable_y()->set_label_string("l (meter)");
+  auto* sl_line = chart->add_line();
+  sl_line->set_label("SL Path");
+  for (const auto& sl_point : sl_frame.sl_path()) {
+    auto* point_debug = sl_line->add_point();
+    point_debug->set_x(sl_point.s());
+    point_debug->set_x(sl_point.l());
+  }
+}
+
+void AddSpeedPlan(
+    const ::google::protobuf::RepeatedPtrField<SpeedPlan>& speed_plans,
+    Chart* chart) {
+  chart->set_title("Speed Plan");
+  auto* options = chart->mutable_options();
+  options->mutable_x()->set_min(0.0);
+  options->mutable_x()->set_max(80.0);
+  options->mutable_x()->set_label_string("s (meter)");
+  options->mutable_y()->set_min(0.0);
+  options->mutable_y()->set_max(50.0);
+  options->mutable_y()->set_label_string("v (m/s)");
```

Comment:
Repeated too many times. Add an anonymous helper function like PopulateChartOptions(x_min, x_max, x_label, y_min, y_max, y_label, *options) and reuse it.

Reply:
> Repeated too many times. Add an anonymous helper function like PopulateChartOptions(x_min, x_max, x_label, y_min, y_max, y_label, *options) and reuse it.

fixed